src/main/java/com/conveyal/analysis/persistence/AnalysisDB.java

```diff
@@ -31,7 +32,11 @@ public AnalysisDB (Config config) {
         // Create codec registry for POJOs
         CodecRegistry pojoCodecRegistry = fromRegistries(
                 MongoClientSettings.getDefaultCodecRegistry(),
-                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+                fromProviders(PojoCodecProvider.builder()
+                        .conventions(Conventions.DEFAULT_CONVENTIONS)
```

Comment:
We should discuss this change and the related `@Bson` annotations

Reply:
Yes, I would like to learn about this change and the Bson annotations.

Reply:
See PojoCodecProvider Builder 
Thinking back on this, I added the @bsonignore annotations in FileStorageFormat to resolve an error, which required registering the Annotation convention.

https://mongodb.github.io/mongo-java-driver/3.11/bson/pojos/#pojos-plain-old-java-objects

https://mongodb.github.io/mongo-java-driver/3.11/javadoc/org/bson/codecs/pojo/Conventions.html#ANNOTATION_CONVENTION

Reply:
We looked at this together today. FileStorageFormat is an enum with fields. Apparently the Mongo driver was trying to serialize these like normal objects, including their field values instead of just serializing the enum value name. There is a canonical instance of the enum class for each enum value, so all references to that enum value have the same values for their fields - it makes no sense for an object codec to save, then try to set the values of these fields. The Mongo documentation linked to above imples this should not be happening: "Enums are fully supported. The POJO Codec uses the name of the enum constant as the property value. This is then converted back into an Enum value by the codec using the static Enum.valueOf method." Maybe it's confused by the fact that these are public fields? But then it would be equally confused by the corresponding property accessor methods on a private field. Maybe something about this POJO Codec Registry configuration are causing it to generate codecs for enums?

Reply:
Digging a bit deeper: PojoCodecProvider.conventions is only used by method createClassModel, which will only overwrite ClassModelBuilder.conventions when non-null;  ClassModelBuilder.conventions is initialized to DEFAULT_CONVENTIONS. So I think `.conventions(Conventions.DEFAULT_CONVENTIONS)` is a no-op. I removed this (reverted the line back to the pre-change state) and when I process an upload, the FileStorageFormat appears to be serialized to Mongo as a simple string representation of the enum's name. I will be on the lookout for some other source of this problem.